16-SpatialRegression.qmd

```diff
@@ -344,6 +344,9 @@ ssre <- predict(GAM_MRF, type = "terms",
 all(sapply(tapply(ssre, list(boston_487$NOX_ID), c),
 		   function(x) length(unique(x)) == 1))
 ```
+<!---
+TRUE expected - I get TRUE locally, your renderings get FALSE (caused by caching?, odd, all determistic steps)
+--->
```

Comment:
I see very small differences:
```
> sapply(tapply(ssre, list(boston_487$NOX_ID), c),
+ 		   function(x) length(unique(x)) == 1)
    1     2     3     5     6     7     8     9    10    11    12    13    14    15    16 
 TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE 
   17    18    19    20    21    22    23    24    25    26    27    28    29    30    31 
 TRUE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE  TRUE 
   32    33    34    35    36    37    38    39    40    41    42    43    44    45    46 
 TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE 
   47    48    49    50    51    52    53    55    56    57    58    59    60    61    62 
 TRUE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE 
   63    64    65    66    67    68    69    70    71    72    74    75    76    77    78 
 TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE 
   79    80    81    82    83    84    85    86    87    88    89    90    91    92    93 
FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE 
   94    95    96 
 TRUE  TRUE  TRUE 
> which(!sapply(tapply(ssre, list(boston_487$NOX_ID), c),
+ 		   function(x) length(unique(x)) == 1))
12 19 28 44 49 62 79 
11 18 27 43 48 60 76 
> s=sapply(tapply(ssre, list(boston_487$NOX_ID), c),
+ 		   function(x) length(unique(x)) == 1)
> which(!s)
12 19 28 44 49 62 79 
11 18 27 43 48 60 76 
> length(s)
[1] 93
> s[which(!s)]
   12    19    28    44    49    62    79 
FALSE FALSE FALSE FALSE FALSE FALSE FALSE 
> ta = (tapply(ssre, list(boston_487$NOX_ID), c))
> ta[[12]]
       78        79        93        94        95        96        97        98        99 
0.0478058 0.0478058 0.0478058 0.0478058 0.0478058 0.0478058 0.0478058 0.0478058 0.0478058 
      100       101       102       103       104       105       106       107       108 
0.0478058 0.0478058 0.0478058 0.0478058 0.0478058 0.0478058 0.0478058 0.0478058 0.0478058 
> unique(ta[[12]])
[1] 0.0478058
> ta[[11]]
        69         71         75         76         77        112        119        120 
0.05483802 0.05483802 0.05483802 0.05483802 0.05483802 0.05483802 0.05483802 0.05483802 
> unique(ta[[11]])
[1] 0.05483802 0.05483802
```
etc. Replacing `x` with `round(x, 8)` gave `TRUE`.

Reply:
OK, could you add the rounding?